yt/geometry/oct_container.pyx

```diff
@@ -728,17 +728,196 @@ cdef class OctreeContainer:
         cdef np.ndarray[np.float64_t, ndim=2] source
         cdef np.ndarray[np.float64_t, ndim=1] dest
         cdef int i
-        cdef np.int64_t local_filled = 0
+
         for key in dest_fields:
             dest = dest_fields[key]
             source = source_fields[key]
             for i in range(levels.shape[0]):
                 if levels[i] != level: continue
-                dest[i + offset] = source[file_inds[i], cell_inds[i]]
-                local_filled += 1
+                if file_inds[i] < 0:
+                    dest[i + offset] = np.nan
+                else:
+                    dest[i + offset] = source[file_inds[i], cell_inds[i]]
+
+    def fill_index(self, SelectorObject selector = AlwaysSelector(None)):
+        """Get the on-file index of each cell"""
+        cdef StoreIndex visitor
+
+        cdef np.int64_t[:, :, :, :] cell_inds
+
+        cell_inds = np.full((self.nocts, 2, 2, 2), -1, dtype=np.int64)
```

Comment:
not used to Cython, so I may be plain wrong, but I read the dtype argument here as completely redundant with the array declaration.

Reply:
Unfortunately it is required to be this way -- we're declaring a cython variable for `cell_inds` but it has to receive it from the python function, which needs to know the dtype.

Reply:
fair enough ! I might end up learning Cython just by reviewing this stuff :)